File: src/pinball/character/Sprint.client.lua
local player = game.Players.LocalPlayer
local character = player.Character or player.CharacterAdded:Wait()
local humanoid = character:WaitForChild("Humanoid")
local UserInputService = game:GetService("UserInputService")
local camera = game.Workspace.CurrentCamera
local runSpeed = 32
local walkSpeed = 16

game:GetService("UserInputService").InputBegan:Connect(function(inputObject)
	if inputObject.KeyCode == Enum.KeyCode.LeftShift then
		local owned = player:FindFirstChild("PinballOwned")
		if owned == nil or owned.Value == nil then
		    humanoid.WalkSpeed = runSpeed
			camera.FieldOfView = 60
		end
	end
end)

game:GetService("UserInputService").InputEnded:Connect(function(inputObject)
	if inputObject.KeyCode == Enum.KeyCode.LeftShift then
		humanoid.WalkSpeed = walkSpeed
		camera.FieldOfView = 70
	end
end)

File: src/StarterPlayer/StarterCharacterScripts/Sprint.client.ts
import { Players, UserInputService } from '@rbxts/services'

const runSpeed = 32
const walkSpeed = 16

const player = Players.LocalPlayer
const character = player.Character || player.CharacterAdded.Wait()[0]
const humanoid = character.WaitForChild('Humanoid') as Humanoid
const camera = game.Workspace.CurrentCamera

UserInputService.InputBegan.Connect((inputObject) => {
  if (inputObject.KeyCode === Enum.KeyCode.LeftShift) {
    const owned = player.FindFirstChild('PinballOwned') as ObjectValue
    if (!owned?.Value) {
      humanoid.WalkSpeed = runSpeed
      if (camera) camera.FieldOfView = 60
    }
  }
})

UserInputService.InputEnded.Connect((inputObject) => {
  if (inputObject.KeyCode === Enum.KeyCode.LeftShift) {
    humanoid.WalkSpeed = walkSpeed
    if (camera) camera.FieldOfView = 70
  }
})
